src/main/java/TaskList.java

```diff
@@ -0,0 +1,158 @@
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * Contains the methods to execute user's commands and store the information while the program runs.
+ */
+public class TaskList {
+
+    public ArrayList<Task> taskList;
+
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    /**
+     *Adds the task to the list.
+     *
+     *  @param t the task to be added.
+     */
+    protected void add(Task t) {
+        taskList.add(t);
+    }
+
+    /**
+     * Prints the addition of a task and number of tasks in the list after addition.
+     */
+    protected void addMessage() {
+        System.out.println(" Got it. I've added this task: ");
+        System.out.println(taskList.get(taskList.size() - 1));
+        if (taskList.size() == 1) {
+            System.out.println("Now you have 1 task in your list.");
+        } else {
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        }
+    }
+
+    /**
+     *Marks the task as done upon user's request and prints the message upon successful completion.
+     *
+     * @param x the position of the task in the list.
+     */
+    protected void markAsDone(int x) {
+        taskList.get(x).markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(taskList.get(x));
+    }
+
+    /**
+     *Deletes the task in the list upon user's request and prints the number of tasks remaining upon deletion.
+     *
+     * @param y the position of the task in the list
+     */
+    protected void deleteTask(int y) {
+        if (taskList.size() == 0) {
+            System.out.println("The task list is empty");
```

Comment:
Perhaps this class should not be printing stuff to the Ui, as it may lead to a difficult time refactoring when u want to change to JavaFX for example.